File: C_Sharp/C_Sharp/domain/validators/UserValidator.cs
namespace C_Sharp.domain
{
    public class UserValidator: IValidator<User>
    {
        public void validate(User entity)
        {
            string errors = "";
            if(entity.UserName == ""){
                errors += "Username is null!\n";
            }
            if(entity.Password == ""){
                errors += "Password is null!\n";
            }
            if(errors != ""){
                throw new ValidationException(errors);
            }
        }
    }
}

File: MPP-Java/src/main/java/domain/validators/UserValidator.java
package domain.validators;

import domain.User;

public class UserValidator implements Validator<User>{
    @Override
    public void validate(User entity) throws ValidationException {
        String errors = "";
        if(entity.getUsername().equals("")){
            errors += "Username is null!\n";
        }
        if(entity.getPassword().equals("")){
            errors += "Password is null!\n";
        }
        if(!errors.equals("")){
            throw new ValidationException(errors);
        }
    }
}
